wordpress-mobile/WordPress-Android
# Followers section in People Management

Closes #4142.

**Decisions**
* I decided to disable swipe to refresh for this feature. Syncing is always an issue for these type of features and to avoid some issues, we implemented a way to remove all the people except for the first page when you first land on the people list. Enabling refresh would make things more complicated and would make our previous decision meaningless. Also, people list should be fairly static and when you go back to My Site and come back to people list, it'll be refreshed.

**Known issues:**
* I recently found some RTL issues while testing the feature: https://cloudup.com/cObeMJEy0yv I'll be looking into this issue, but thought it shouldn't hold the code review.
* There is a possible design change for the subscriber date position, currently waiting feedback from @rickybanister
* While testing a large list of followers, I noticed that when you go into detail and come back you lose your scroll position. I've tried many times to reproduce this, but was unable to do so. I'll keep an eye on this one and please let me know if you notice anything similar.

**Questions**
* If there is an error, we currently show a toast as well as set the empty view message to that error. I am not sure if the duplicate approach is good here. Should we remove that toast message?
* I feel like the loading progress bar, for infinite scroll, should be centered. However, that component is used on other pages as well, so we should make a decision on whether we should move it for all those pages or keep it as is.

**To test:**
* Go into the People section from My Site page and check your followers & email followers
* It'd be great if we install a version from `develop` first and then update it from this branch to test the migration. The migration is pretty minimal, but it's a good check. Make sure you visit the people list when you first install it from `develop` so the db will have some data to work with.
* There is a lot of state management going on, so it'd be good to test network availability, empty lists for followers & email followers, loading the list for the first time, loading more followers etc.
* If you have a site with many followers, it's a good test as well to make sure our infinite scroll is working correctly.
* Go into the detail page of a follower and try to remove her.
* While trying to add an email follower to your site for test purposes, keep in mind that the email can not belong to a WordPress.com user. You can always use the `youremail+something@example.com` gmail trick for this.

/cc @hypest, @astralbodies 

@mzorz as we discussed on Slack a while ago, I am pinging you for review since you built the original version of `FilteredRecyclerView` and most of this PR is about that. Do let me know if you have any questions about the specifics of People Management and if there is anything not clear at first glance so I can add comments accordingly. Thanks!

Needs review: @mzorz
